// bytom-sdk/testCompiled/testHelpers.js
'use strict';

var bytom = require('../dist/index.js');
var uuid = require('uuid');
var url = 'http://localhost:9888';
var accessToken = '';

var client = new bytom.Client(url, accessToken);

var balanceByAssetAlias = function balanceByAssetAlias(balances) {
  var res = {};
  return Promise.resolve(balances).then(function (balance) {
    balance.forEach(function (item) {
      res[item.sumBy.assetAlias] = item.amount;
    });
    return res;
  });
};

var createAccount = function createAccount() {
  var account = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 'account';

  return client.keys.listAll().then(function (keys) {
    return client.accounts.create({
      alias: account + '-' + uuid.v4(),
      root_xpubs: [keys[0].xpub],
      quorum: 1
    });
  });
};

var createAccountReciever = function createAccountReciever() {
  var accountAlias = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 'account';

  return client.accounts.createReceiver({
    account_alias: accountAlias
  });
};

var createAsset = function createAsset() {
  var asset = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 'asset';

  return client.keys.listAll().then(function (keys) {
    return client.assets.create({
      alias: asset + '-' + uuid.v4(),
      definition: {
        decimals: 8,
        description: {},
        name: asset + '-' + uuid.v4(),
        symbol: asset + '-' + uuid.v4()
      },
      root_xpubs: [keys[0].xpub],
      quorum: 1
    });
  });
};

var buildSignSubmit = function buildSignSubmit(buildFunc, optClient, password) {
  var c = optClient || client;
  return c.transactions.build(buildFunc).then(function (tpl) {
    return c.transactions.sign({
      transaction: tpl,
      password: password
    });
  }).then(function (tpl) {
    return c.transactions.submit(tpl.transaction.raw_transaction);
  });
};

module.exports = {
  // balanceByAssetAlias,
  client: client,
  createAccount: createAccount,
  createAsset: createAsset,
  createAccountReciever: createAccountReciever,
  buildSignSubmit: buildSignSubmit
};
