// bytom-sdk/dist/errors.js
'use strict';

var lib = {
  create: function create(type, message) {
    var props = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : {};

    var err = void 0;
    if (props.body) {
      err = lib.newBatchError(props.body, props.requestId);
    } else {
      err = new Error(message);
    }

    err = Object.assign(err, props, {
      bytomClientError: true,
      type: type
    });
    return err;
  },

  isbytomError: function isbytomError(err) {
    return err && !!err.bytomClientError;
  },

  isBatchError: function isBatchError(v) {
    return v && v.code && !v.stack;
  },

  newBatchError: function newBatchError(body) {
    var requestId = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : false;

    var err = new Error(lib.formatErrMsg(body, requestId));
    err.code = body.code;
    err.bytomMessage = body.msg;
    err.detail = body.error_detail;
    return err;
  },

  formatErrMsg: function formatErrMsg(body) {
    var tokens = [];

    if (typeof body.code === 'string' && body.code.length > 0) {
      tokens.push('Code: ' + body.code);
    }

    tokens.push('Message: ' + body.msg);

    if (typeof body.error_detail === 'string' && body.error_detail.length > 0) {
      tokens.push('Detail: ' + body.error_detail);
    }

    return tokens.join(' ');
  },

  types: {
    FETCH: 'FETCH',
    CONNECTIVITY: 'CONNECTIVITY',
    JSON: 'JSON',
    UNAUTHORIZED: 'UNAUTHORIZED',
    NOT_FOUND: 'NOT_FOUND',
    BAD_REQUEST: 'BAD_REQUEST',
    SERVER_ERROR: 'SERVER_ERROR'
  }
};

module.exports = lib;
